default/src/components/signer.js

```diff
@@ -120,6 +128,139 @@ const GetTxParams = () => {
   );
 };
 
+const SendASA = () => {
+  let asaData = YAMLData.default.default.asa;
+  let deployedASAName;
+  let deployedASAId;
+  let asaId;
+  for (const [key, value] of Object.entries(asaData)) {
+    deployedASAName = key;
+    deployedASAId = value.assetIndex;
+  }
+
+  const [result, setResult] = useState('');
+
+  const action1 = useCallback(async () => {
+    asaId = deployedASAId;
+  });
+
+  const action2 = useCallback(async () => {
+    asaId = prompt('Please enter ASA ID');
+  });
+
+  const transfer = useCallback(async () => {
+    try {
+      let receiver = document.getElementById('recvAddr').value;
+      if (receiver === '') {
+        return "Please enter receiver's address.";
+      }
+      let sender = document.getElementById('sndrAddr').value;
+      if (sender === '') {
+        return "Please enter sender's address.";
+      }
+      let amount = document.getElementById('amount').value;
+      if (amount === '') {
+        return 'Please enter amount.';
+      }
+      if (!Number(asaId)) {
+        return 'Entered ASA ID is not a number.';
+      }
+      if (!Number(amount)) {
+        return 'Entered amount is not a number.';
+      }
+      const r = await ASATransfer(
+        Number(asaId),
+        sender,
+        receiver,
+        Number(amount)
+      );
+      setResult(r);
+    } catch (e) {
+      console.error(e);
+      setResult(JSON.stringify(e, null, 2));
+    }
+  }, []);
+  return (
+    <div>
+      <Typography variant="h5" style={{ margin: { spacing } }}>
+        ASA Transfer using AlgoSigner
+      </Typography>
+      <div>
+        <PopupState variant="popover" popupId="popup-menu">
+          {popupState => (
+            <React.Fragment>
+              <Button
+                variant="contained"
+                color="secondary"
+                size="large"
+                {...bindTrigger(popupState)}
```

Comment:
Can you please explain what *{...bindTrigger(popupState)}* does ?

Reply:
It handles the state of the ASA Selection Button.